rasa/core/training/interactive.py

```diff
@@ -803,14 +803,38 @@ def _collect_actions(events: List[Dict[Text, Any]]) -> List[Dict[Text, Any]]:
                 f.write("\n" + tracker.export_stories(SAVE_IN_E2E))
 
 
+def _retrieve_intents_from_domain(serialised_domain: Dict[Text, Any]) -> List[Text]:
+    """Retrieve the intents from the serialised domain"""
+
+    intents = [next(iter(i)) for i in (serialised_domain.get("intents") or [])]
+    return intents
+
+
+def _remove_intent_payload_input(
+    msgs: List[Message], intents: List[Text]
+) -> List[Message]:
+    """Filter messages removing intents before write nlu to file"""
+
+    filtered_messages = []
+    for msg in msgs:
+        if not (msg.text.startswith(INTENT_MESSAGE_PREFIX) and msg.text[1:] in intents):
```

Comment:
do we actually need to check if the intent is in the domain? I think there shouldn't be an issue if we always disregard any messages that start with `INTENT_MESSAGE_PREFIX `. 

this would make the change simpler. what do you think?

Reply:
If we can always disregard messages that start with `INTENT_MESSAGE_PREFIX`, the implementation will be a lot simpler. As I can see, there would be 3 major changes in the implementation:

Reply:
If

Reply:
If we can always disregard messages that start with `INTENT_MESSAGE_PREFIX` the implementation would be a lot simpler. I can see 3 major changes in the implementation:
- The function `_retrieve_intents_from_domain` would no long be necessary;
- The extra parameter `serialised_domain` passed to the function `_write_nlu_to_file` could be removed;
- The if on line 820 would only need to check if the message does not start with `INTENT_MESSAGE_PREFIX`.

Reply:
alright let's do that - I double-checked and I can't see a case where we would need to add a message starting with `/` to the training data.